src/rdkafka_txnmgr.c

```diff
@@ -2588,49 +2631,6 @@ static rd_kafka_op_res_t rd_kafka_txn_op_abort_transaction(rd_kafka_t *rk,
                 goto done;
         }
 
-        if (rk->rk_eos.txn_requires_epoch_bump ||
-            rk->rk_eos.idemp_state != RD_KAFKA_IDEMP_STATE_ASSIGNED) {
-                /* If the underlying idempotent producer's state indicates it
```

Comment:
Please do not remove comments as they're needed to explain the why of the code, and we're not really changing the behaviour so this comment is still mostly correct.